src/DotNetWorker/Diagnostics/WorkerLoggerT.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Functions.Worker.Diagnostics
+{
+    internal class WorkerLogger<T> : ILogger<T>
+    {
+        private readonly WorkerLogger _inner;
+        private static Assembly _thisAssembly = typeof(WorkerLogger<>).Assembly;
+
+        public WorkerLogger(ILoggerFactory loggerFactory)
+        {
+            bool isSystemAssembly = typeof(T).Assembly == _thisAssembly;
```

Comment:
I'm not really loving this. It means that:
1. If someone were to create a `ILogger<HttpRequestData>` (or any of our types) for some reason, it'd suddenly be treated as a system log. 
2. If someone were to overwrite our `ILogger<>` service registration, we'd stop having system logs. We want to get out of the way of this as much as possible as it's bitten us before.
 
I have a different plan for this that'll be a little overhead for us logging internally but would eliminate both of these issues. Will update this PR.